src/GraphQL/Types/Composite/AutoRegisteringInterfaceGraphType.cs

```diff
@@ -0,0 +1,154 @@
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace GraphQL.Types
+{
+    /// <summary>
+    /// Allows you to automatically register the necessary fields for the specified type.
+    /// Supports <see cref="DescriptionAttribute"/>, <see cref="ObsoleteAttribute"/>, <see cref="DefaultValueAttribute"/> and <see cref="RequiredAttribute"/>.
+    /// Also it can get descriptions for fields from the XML comments.
+    /// </summary>
+    public class AutoRegisteringInterfaceGraphType<TSourceType> : InterfaceGraphType<TSourceType>
+    {
+        private readonly Expression<Func<TSourceType, object?>>[]? _excludedProperties;
+
+        /// <summary>
+        /// Creates a GraphQL type from <typeparamref name="TSourceType"/>.
+        /// </summary>
+        public AutoRegisteringInterfaceGraphType() : this(null) { }
+
+        /// <summary>
+        /// Creates a GraphQL type from <typeparamref name="TSourceType"/> by specifying fields to exclude from registration.
+        /// </summary>
+        /// <param name="excludedProperties"> Expressions for excluding fields, for example 'o => o.Age'. </param>
+        public AutoRegisteringInterfaceGraphType(params Expression<Func<TSourceType, object?>>[]? excludedProperties)
+        {
+            _excludedProperties = excludedProperties;
+            Name = typeof(TSourceType).GraphQLName();
+            ConfigureGraph();
+            foreach (var fieldType in ProvideFields())
+            {
+                _ = AddField(fieldType);
+            }
+        }
+
+        /// <inheritdoc cref="AutoRegisteringObjectGraphType{TSourceType}.ConfigureGraph"/>
```

Comment:
All of the comments are written suitably for interfaces just as well as classes, so to deduplicate the comments, I just pointed everything except the constructors to the other class.